helix-term/src/handlers/completion/path.rs

```diff
@@ -0,0 +1,194 @@
+use std::{
+    borrow::Cow,
+    fs,
+    path::{Path, PathBuf},
+    str::FromStr as _,
+    sync::{
+        atomic::{AtomicBool, Ordering},
+        Arc,
+    },
+};
+
+use futures_util::{future::BoxFuture, FutureExt as _};
+use helix_core as core;
+use helix_core::Transaction;
+use helix_stdx::path::{self, canonicalize, fold_home_dir, get_path_suffix};
+use helix_view::Document;
+use url::Url;
+
+use super::item::CompletionItem;
+
+pub(crate) fn path_completion(
+    cursor: usize,
+    text: core::Rope,
+    doc: &Document,
+    cancel: Arc<AtomicBool>,
+) -> Option<BoxFuture<'static, anyhow::Result<Vec<CompletionItem>>>> {
+    if !doc.path_completion_enabled() {
+        return None;
+    }
+
+    let cur_line = text.char_to_line(cursor);
+    let start = text.line_to_char(cur_line).max(cursor.saturating_sub(1000));
+    let line_until_cursor = text.slice(start..cursor);
+
+    let (dir_path, typed_file_name) =
+        get_path_suffix(line_until_cursor, false).and_then(|matched_path| {
+            let matched_path = Cow::from(matched_path);
+            let path: Cow<_> = if matched_path.starts_with("file://") {
+                Url::from_str(&matched_path)
+                    .ok()
+                    .and_then(|url| url.to_file_path().ok())?
+                    .into()
+            } else {
+                Path::new(&*matched_path).into()
+            };
+            let path = path::expand(&path);
+            let parent_dir = doc.path().and_then(|dp| dp.parent());
+            let path = match parent_dir {
+                Some(parent_dir) if path.is_relative() => parent_dir.join(&path),
+                _ => path.into_owned(),
+            };
+            let ends_with_slash = matches!(matched_path.as_bytes().last(), Some(b'/' | b'\\'));
```

Comment:
This may have been my mistake but I think we want to only set ends with slash to true for backslash if we are on windows (so for windows we want to check for both forward and reverse slash and for unix we only want to check for forward)